### PR TITLE
chore(server): Increase domain package unit test coverage (asset and integrationapi)

### PR DESCRIPTION
**Pull request by jasonkarel**
# Overview
Increase unit testing coverage in asset (85% to 99%) and integrationapi (2% to 51.5%) packages
## What I've done
added more ut to cover codes in asset and integrationapi packages
## How I tested
run package coverage test
## Which point I want you to review particularly
test cases of asset and integrationapi packages.
## Summary by CodeRabbit

- **New Features**
	- Introduced new test functions to validate various functionalities, including asset creation, upload handling, and project instantiation.
	- Enhanced test coverage for edge cases, including handling of `nil` inputs and various status conditions.

- **Bug Fixes**
	- Adjusted existing tests to improve error handling and ensure proper assertions for edge cases.

- **Tests**
	- Expanded the test suite with multiple new functions and refined existing ones for better clarity and coverage.

### CODE REVIEW

File: server/pkg/asset/upload_builder_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		},
+	}
+
+	timeNow := time.Now
+	assert.Equal(t, ubWithData, ubWithData.UUID("1"))
+	assert.Equal(t, ubWithData, ubWithData.Project(projectID))
+	assert.Equal(t, ubWithData, ubWithData.FileName("file.test"))
+	assert.Equal(t, ubWithData, ubWithData.ContentLength(int64(1)))
+	assert.Equal(t, ubWithData, ubWithData.ExpiresAt(timeNow()))
+	assert.Equal(t, ubWithData.u, ubWithData.Build())
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/asset/upload_builder_test.go:33]: ignoring this since it returns builder object.

File: server/pkg/asset/builder_test.go [hunk 2]
```diff
@@ -53,6 +54,7 @@ func TestBuilder_Build(t *testing.T) {
 				uuid:                    "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
 				thread:                  thid,
 				archiveExtractionStatus: lo.ToPtr(ArchiveExtractionStatusPending),
+				flatFiles:               false,
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/asset/builder_test.go:57]: the code deosnt handle if flatfiles is true/ false.

File: server/pkg/integrationapi/value_test.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+		{
+			name:     "Valid ValueTypeRichText",
+			input:    lo.ToPtr(ValueTypeRichText),
+			expected: value.TypeRichText,
+		},
+		{
+			name:     "Valid ValueTypeMarkdown",
+			input:    lo.ToPtr(ValueTypeMarkdown),
+			expected: value.TypeMarkdown,
+		},
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/integrationapi/value_test.go:45]: will edit

File: server/pkg/integrationapi/comment_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			},
+		},
+		{
+			name:  "Integration author",
+			input: cIntegration,
+			expected: &Comment{
+				Content:    lo.ToPtr("test"),
+				CreatedAt:  lo.ToPtr(cIntegration.CreatedAt()),
+				Id:         cIntegration.ID().Ref(),
+				AuthorType: lo.ToPtr(Integrtaion),
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/integrationapi/comment_test.go:49]: will fix the typo issue later as it will change the non unit testing code. the task in here is the unit testing code

File: server/pkg/integrationapi/project_test.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := NewProject(p1)
+			assert.Equal(t, tt.want, result)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/integrationapi/project_test.go:41]: will fix

File: server/pkg/integrationapi/project_test.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+				Id:          p1.ID().Ref(),
+				WorkspaceId: p1.Workspace().Ref(),
+				Alias:       lo.ToPtr(p1.Alias()),
+				Name:        lo.ToPtr(p1.Name()),
+				Description: lo.ToPtr(p1.Description()),
+				CreatedAt:   lo.ToPtr(p1.CreatedAt()),
+				UpdatedAt:   lo.ToPtr(p1.UpdatedAt()),
+			},
+		},
+	}
```
[the remark this answers is not in the document]
Comment (jasonkarel) [reply server/pkg/integrationapi/project_test.go:34]: will skip as the code doesnt handle such case

File: server/pkg/integrationapi/asset_test.go [hunk 7]
```diff
@@ -46,4 +51,183 @@ func TestToAssetFile(t *testing.T) {
...
+		{
+			name:     "Status pending",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatusPending),
+			expected: lo.ToPtr(AssetArchiveExtractionStatus("pending")),
+		},
+		{
+			name:     "Unknown status",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatus("unknown")),
+			expected: nil,
+		},
```
Comment (yk-eukarya) [inline server/pkg/integrationapi/asset_test.go:156]:
```suggestion
		{
			name:     "Status done",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusDone),
			expected: lo.ToPtr(integrationapi.Done),
		},
		{
			name:     "Status failed",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusFailed),
			expected: lo.ToPtr(integrationapi.Failed),
		},
		{
			name:     "Status in progress",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusInProgress),
			expected: lo.ToPtr(integrationapi.InProgress),
		},
		{
			name:     "Status pending",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusPending),
			expected: lo.ToPtr(integrationapi.Pending),
		},
		{
			name:     "Unknown status",
			input:    lo.ToPtr(asset.ArchiveExtractionStatus("unknown")),
			expected: nil,
		},
```
use the defined constants

Comment (jasonkarel) [reply server/pkg/integrationapi/asset_test.go:156]: will fix

File: server/pkg/integrationapi/item_export_test.go [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := FeatureCollectionFromItems(tt.inputVer, tt.inputSchema)
+			assert.Equal(t, tt.expected, result, "FeatureCollectionFromItems() expected %v but got %v", tt.expected, result)
+			if (err != nil) == tt.expectError {
+				fmt.Printf("Error is not matched")
+			}
+		})
```
[the remark this answers is not in the document]
Comment (yk-eukarya) [reply server/pkg/integrationapi/item_export_test.go:141]: @jasonkarel plz fix this one

**Review by yk-eukarya — COMMENTED** [review]
Please use the parallel test whenever possible

File: server/pkg/asset/upload_builder_test.go [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+			uuid:          "1",
+			project:       projectID,
+			fileName:      "file.test",
+			contentLength: int64(1),
+			expiresAt:     time.Now(),
+		},
+	}
+
+	timeNow := time.Now
+	assert.Equal(t, ubWithData, ubWithData.UUID("1"))
```
[the remark this answers is not in the document]
Comment (yk-eukarya) [reply server/pkg/asset/upload_builder_test.go]: @jasonkarel plz use util.MockNow from reearthx

File: server/pkg/integrationapi/condition_test.go [hunk 10]
```diff
@@ -0,0 +1,708 @@
...
+		expected view.NullableOperator
+	}{
+		{Empty, view.NullableOperatorEmpty},
+		{NotEmpty, view.NullableOperatorNotEmpty},
+		{ConditionNullableOperator("99"), ""}, // Test for default case
+	}
+
+	for _, test := range tests {
+		result := test.input.Into()
+		if result != test.expected {
```
Comment (yk-eukarya) [inline server/pkg/integrationapi/condition_test.go:296]: use assert.Equal

File: server/pkg/integrationapi/condition_test.go [hunk 11]
```diff
@@ -0,0 +1,708 @@
...
+		{IncludesAny, view.MultipleOperatorIncludesAny},
+		{NotIncludesAny, view.MultipleOperatorNotIncludesAny},
+		{IncludesAll, view.MultipleOperatorIncludesAll},
+		{NotIncludesAll, view.MultipleOperatorNotIncludesAll},
+		{ConditionMultipleOperator("99"), ""}, // Test for default case
+	}
+
+	for _, test := range tests {
+		result := test.input.Into()
+		if result != test.expected {
```
Comment (yk-eukarya) [inline server/pkg/integrationapi/condition_test.go:316]: use asset.Equal

File: server/pkg/integrationapi/condition_test.go [hunk 12]
```diff
@@ -0,0 +1,708 @@
...
+			expected: view.FieldSelector{
+				Type: view.FieldTypeId,
+				ID:   fieldType.FieldId,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		result := test.input.Into()
+		if result != test.expected {
```
Comment (yk-eukarya) [inline server/pkg/integrationapi/condition_test.go:347]: use asset.Equal

File: server/pkg/integrationapi/condition_test.go [hunk 13]
```diff
@@ -0,0 +1,708 @@
...
+			condition: &Condition{},
+			want:      nil,
+		},
+	}
+	for _, test := range tests {
+		result := test.condition.Into()
+		if !assert.Equal(t, result, test.want) {
+			t.Errorf("expected %+v but got %+v", test.want, result)
+		}
+	}
```
Comment (yk-eukarya) [inline server/pkg/integrationapi/condition_test.go:707]:
```suggestion
	for _, tc := range tests {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()
			assert.Equal(t, tt.want, tc.condition.Into())
		})
	}
```

**Review by yk-eukarya — APPROVED** [review]
